flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/StandaloneFlinkService.java

```diff
@@ -171,11 +170,6 @@ protected void deleteClusterInternal(ObjectMeta meta, boolean deleteHaConfigmaps
 
     @Override
     public boolean scale(ObjectMeta meta, JobSpec jobSpec, Configuration conf) {
-        if (conf.get(JobManagerOptions.SCHEDULER_MODE) == null) {
```

Comment:
This check was supposed to check whether reactive scheduler is enabled for applications, we can’t simply remove it.

Maybe we are lacking test coverage for this…

In any case it checks not null which already looks incorrect. @morhidi you worked on this in the past what do you think?

Reply:
Even if there is no scheduler mode enabled , i.e if the scheduler mode is null, still we should be able to scale the cluster i.e even if the adaptive or reactive mode is not enabled and only scaling change should allow to only scale. But let me know your thoughts @gyfora , @morhidi

Reply:
In this case, we are calculating the desired replicas as 
```
var desiredReplicas =
                conf.get(StandaloneKubernetesConfigOptionsInternal.KUBERNETES_TASKMANAGER_REPLICAS);
```
Which is actually ```kubernetes.internal.taskmanager.replicas``` which is created whenever there is a CR update while building the task manager. Here, this value would be equal to the following:
1) If taskmanager replicas are present, KUBERNETES_TASKMANAGER_REPLICAS = task_manager_replicas configured. [Code Reference to 1](https://github.com/apache/flink-kubernetes-operator/blob/dc173cbccd5f9a90fe573ce01790684253dcc0a9/flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java#L233) 
2) Else, if a job is running, it calculates the parallelism = task_slots * task_manager_replicas or parallelism configured ( if no TM replicas ) and [Code reference to 2 ](https://github.com/apache/flink-kubernetes-operator/blob/dc173cbccd5f9a90fe573ce01790684253dcc0a9/flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java#L247)
```KUBERNETES_TASKMANAGER_REPLICAS =  (parallelism + taskSlots - 1) / taskSlots;```
[Internal task manager calc for 2 ]( https://github.com/apache/flink-kubernetes-operator/blob/dc173cbccd5f9a90fe573ce01790684253dcc0a9/flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java#L174)

We'll always fall to 1st category as the user has updated the task manager replicas manually. 


@gyfora , @morhidi  if you think, we have other scenarios which might affect it,let me know, I'll create another diffType ( SCALE_REPLICAS ) which will be used only by task manager replicas change and add a separate logic to scale based on the desired TM replicas and not ```kubernetes.internal.taskmanager.replicas```

Reply:
That's fine but what I am saying is that the StandaloneFlinkService#scale method is used by both Application and Session cluster deployments. You cannot simply scale an application cluster by adding more replicas unless reactive scheduler is configured for the job. 

So the check that you removed is neeeded for the application logic to work. 
And it should be improved to:
```
if (conf.get(JobManagerOptions.SCHEDULER_MODE) == REACTIVE) {
            return false;
        }
```

Reply:
As you can see your change actually breaks the test case for this:
```
StandaloneFlinkServiceTest.testReactiveScale:180 expected: <false> but was: <true>
```

You need to add new tests in this class to make sure this works well for sessions too and still preserve the existing behaviour for Application deployments

Reply:
Understood your concern, will make changes to make it work for both session and application mode

Reply:
Addressed the comment where it scales in appmode only if parallelism changes, and if there is a replica change when the reactive scheduler is not enabled, the scaling is not successful. Added test cases as well in both session and app mode for different scenarios. 
Currently scaling can happen due to the following in standalone clusters :
1) Session mode - replica change
2) App mode enabled with scheduler configured